semeio/workflows/localisation/localisation_config.py

```diff
@@ -0,0 +1,311 @@
+    the covariance will be reduced by this factor:
+
+    new_cov(field(i,j,k)) = orig_cov(field(i,j,k), obs) * scaling_factor
+
+    For some of methods for calculating scaling factors, a reference point,
+    typically a well location, must be specified. For other methods any reference
+    point is not used.
+
+    """
+
+    name: str
+    obs_group: ObsConfig
+    param_group: ParamConfig
+    ref_point: Optional[conlist(float, min_items=2, max_items=2)]
+    field_scale: Optional[Union[GaussianConfig, ExponentialConfig]]
+    surface_scale: Optional[Union[GaussianConfig, ExponentialConfig]]
+    obs_context: list
+    params_context: list
+
+    @root_validator(pre=True)
+    def inject_context(cls, values: Dict) -> Dict:
+        values["obs_group"]["context"] = values["obs_context"]
+        values["param_group"]["context"] = values["params_context"]
+        return values
+
+    @validator("field_scale", pre=True)
+    def validate_field_scale(cls, value):
+        """
+        To improve the user feedback we explicitly check
+        which method is configured and bypass the Union
+        """
+        if isinstance(value, BaseModel):
+            return value
+        if not isinstance(value, dict):
+            raise ValueError("value must be dict")
+        method = value.get("method")
+        _valid_methods = {
+            "gaussian_decay": GaussianConfig,
+            "exponential_decay": ExponentialConfig,
+        }
+        if method in _valid_methods.keys():
+            return _valid_methods[method](**value)
+        else:
+            raise ValueError(
+                f"Unknown method: {method}, valid methods are: {_valid_methods.keys()}"
+            )
+
+    @root_validator()
+    def valid_ref_point_and_scale(cls, values: Dict) -> Dict:
+        field_scale = values.get("field_scale", None)
+        surface_scale = values.get("surface_scale", None)
+        ref_point = values.get("ref_point")
+        if (field_scale or surface_scale) and not ref_point:
+            raise ValueError("If scaling surface or field ref_point must be provided")
+        return values
+
+    @validator("surface_scale", pre=True)
+    def validate_surface_scale(cls, value):
+        """
+        To improve the user feedback we explicitly check
+        which method is configured and bypass the Union
+        """
+        if isinstance(value, BaseModel):
+            return value
+        if not isinstance(value, dict):
+            raise ValueError("value must be dict")
+
+        # String with relative path to surface files relative to config path
+        key = "surface_file"
+        if key not in value.keys():
+            raise ValueError(f"Missing keyword: '{key}' in keyword: 'surface_scale' ")
+
+        filename = value[key]
+        # Check  that the file exists
+        if not os.path.exists(filename):
+            raise IOError(f"File for surface: {filename} does not exist.")
```

Comment:
Because this is part of the pydentic schema you want to raise `ValueError` or ` TypeError` so pydantic re-raises as validation error (`ValueError` in this case)

Reply:
OK, I change this